src/keyboard.cpp

```diff
@@ -123,11 +123,38 @@ void keyboard::show()
 	QWidget::show();
 	//Moving after showing because window height isn't set until show()
 	this->move(0,QApplication::desktop()->screenGeometry().height() - height());
+    QTimer::singleShot(1, this, SLOT(selectAllInFocusedWidget()));
+}
 
+void keyboard::selectAllInFocusedWidget(){
+    emit codeGenerated(KC_RIGHT); //to deselect any text that might already be selected
+    emit characterGenerated(QChar('a')); //insert arbitrary char to have selectAll() have any effect
 
-}
+    QString senderClass = lastFocusedWidget->metaObject()->className();
```

Comment:
I don't recommend this approach of determining the class by name and then casting to the class structure in a big block of if/else statements. Presumeably the selectAll() function is inherited from some common ancestor of the various QXxxEdit classes, and we simply need to get that common ancestor's class.

This kind of 'if/else block' scales poorly, since you need to updated it for every new class that comes along, and tends to be prone to bugs.

Reply:
QLineEdit and QTextEdit inherit from QWidget, which does not have selectAll(). https://doc.qt.io/archives/qt-4.8/qlineedit.html#selectAll
At least the spinbox classes both inherit from QAbstractSpinbox, which does have selectAll().
So, why don't we use a switch statement instead? We could use the same code for both the spinbox classes.